### DISCUSSION TITLE
[Half-joking request] Add Neofetch to Terminal

### DISCUSSION THREAD

**Discussion by wertercatt** [DustinBrett/daedalOS#61] [Ideas]
Then you can post it on /r/unixporn

**Comment by DustinBrett** [comment]
Thanks for the idea! I like it!

**Reply by hello-smile6** [reply]
This'd be fun... Could I?

**Reply by DustinBrett** [reply]
Feel free to submit a PR :-) I'd be interested to see it. Indeed sounds fun.

**Reply by hello-smile6** [reply]
It'd probably use Buildroot.

**Reply by DustinBrett** [reply]
I'm always interested in seeing contribution. I noticed https://github.com/golota60/yayfetch is a nice JS version.

**Reply by hello-smile6** [reply]
https://github.com/golota60/yayfetch

**Reply by hello-smile6** [reply]
@DustinBrett That's for node, it seems.

**Reply by DustinBrett** [reply]
Its just to get started. Just an idea.

**Reply by hello-smile6** [reply]
https://github.com/hello-smile6/neofetch-js
I'll try to make something similar that takes parameters from whatever it's being used in, so people can use it for anything, not just for websites. Should I define ANSI escape sequences as constants?

**Reply by hello-smile6** [reply]
@DustinBrett How am I doing on it? I haven't gotten output done yet, but I should have an xtermjs web demo in a few minutes.

**Reply by hello-smile6** [reply]
@DustinBrett @wertercatt https://neofetch-js.vercel.app/

**Reply by DustinBrett** [reply]
Looks interesting for sure.

**Reply by hello-smile6** [reply]
> Looks interesting for sure.

If you'd like to use it, I'll make a PR.

**Reply by hello-smile6** [reply]
I'd have to figure out how, though.

**Reply by DustinBrett** [reply]
Feel free to throw up a PR, but I cant be sure when I get to look at it and I had quite a different idea for Neofetch I think in that I'd rather it be an icon of my OS, it cant be windows like logo, and the info could match daedalOS and not the persons user agent which can be tricky.

**Reply by hello-smile6** [reply]
It almost has support for that. You can already specify a custom OS, it just needs a way to provide arbitrary text for the logo.

**Reply by Cveinnt** [reply]
Late to the party but is there an update on this? For now I am using a fake fetch written in python to mimic neofetch.

**Reply by DustinBrett** [reply]
It's doable, but to not have it be fake I'd need to spend a bit more time on the specs/stats info and the ascii logo. For sure one day...

**Comment by DustinBrett** [comment]
If I were to add this I was thinking of going for as realistic as possible. So perhaps I would convert the start menu icon into ascii art for the left image and then I would need to try and find useful information to display based on the fake and real os and anything else find-out-able.

**Reply by wertercatt** [reply]
Maybe you could print the user agent as the 'Host'

**Reply by DustinBrett** [reply]
Ya that would be good. I do something similar if you type whoami in the terminal.

https://github.com/DustinBrett/daedalOS/blob/a4d78d6712101baaea1f89890635e6ef5e4b8bac/components/apps/Terminal/useCommandInterpreter.ts#L594

**Comment by DustinBrett** [comment]
@wertercatt @Cveinnt @hello-smile6 I've finally taken the time to add neofetch (https://github.com/DustinBrett/daedalOS/commit/fc0da5d535c6c0d5a700e5f991b40e46aa7db954 & https://github.com/DustinBrett/daedalOS/commit/fdf4c3ef9d17a05881a5a30675b5d4c5288492cd) to Terminal. The information it displays should be mostly dynamic.

![image](https://user-images.githubusercontent.com/16656318/214154675-e73bb8d0-697f-454f-9251-ab3be3858fef.png)
